### PR TITLE
Add mo-kurobako benchmark to CI

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation
Currently, optuna has a benchmark `mo-kurobako` but cannot be run with GitHub Actions (cf. #3245). We add a YAML file to enable that.

## Description of the changes
Add `performance-benchmarks-mo-kurobako.yml`.

### CODE REVIEW

**Comment by contramundum53** [comment]
@toshihikoyanase @knshnb Could you review this PR?

**Comment by knshnb** [comment]
Thank you for the PR! I confirmed that it works on GitHub Actions.

As a side note, I found several issues in the output files.
- `report.md` is not informative (all the values in "Best" and "AUC" are NaN).
image

- WFG1 might not be working correctly.
image

These issues do not seem to be caused by this PR, so you don't need to address them in this PR.

**Comment by knshnb** [comment]
I thought it is helpful to add a brief explanation of output files including the following two points.
- It outputs different files for different samplers in the multi-objective benchmark.
- Currently, `report.md` is not informative.

File: .github/workflows/performance-benchmarks-mo-kurobako.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      uses: actions/cache@v2
+      with:
+        path: ./fcnet_tabular_benchmarks
+        key: hpobench-dataset
+
+    - name: Download hpobench dataset
+      if: steps.cache-hpobench-dataset.outputs.cache-hit != 'true'
+      run: |
+        wget http://ml4aad.org/wp-content/uploads/2019/01/fcnet_tabular_benchmarks.tar.gz
+        tar xf fcnet_tabular_benchmarks.tar.gz
```
Comment (HideakiImamura) [inline .github/workflows/performance-benchmarks-mo-kurobako.yml:89]: We don't use HPO datset for the multi-objective kurobako benchmark. Can be removed.
```suggestion
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have some comments. PTAL.

**Comment by contramundum53** [comment]
Thanks for your comments! I applied your suggestions.

**Review by HideakiImamura — APPROVED** [review]
Thanks, LGTM!

**Review by knshnb — APPROVED** [review]
LGTM! Let me merge this PR.

File: .github/workflows/performance-benchmarks-mo-kurobako.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        key: kurobako-0-2-9
+
+    - name: Install kurobako CLI
+      if: steps.cache-kurobako.outputs.cache-hit != 'true'
+      run: |
+        curl -L https://github.com/optuna/kurobako/releases/download/0.2.9/kurobako-0.2.9.linux-amd64 -o kurobako
+
+        chmod +x kurobako
+        ./kurobako -h
+
```
Comment (knshnb) [inline .github/workflows/performance-benchmarks-mo-kurobako.yml:77]:
```suggestion
```

Comment (knshnb) [reply .github/workflows/performance-benchmarks-mo-kurobako.yml:77]: Two blank lines are generated by the previous change.

**Review by knshnb — COMMENTED** [review]
Oh, sorry, but let me suggest a very small modification.

**Comment by HideakiImamura** [comment]
Let me leave some todo comments for the multi-objective benchmark.
- We should consider other performance metrics than pareto front plots, and implement it to the kurobako or benchmark script.
- We should fix WFG1. Maybe, we should add some tests for the WFG test problem.
- We should fix the `report.md` of the kurobako for multu-objective optimization.
- We should fix the pareto front plots to only consider the trials in one study. Currently, it gathers all studies for each setting.

**Comment by HideakiImamura** [comment]
I will create an issue about that.

**Comment by HideakiImamura** [comment]
I created some issues for follow-ups. #3716, #3717, #3718, #3719

**Comment by knshnb** [comment]
LGTM!

**Pull request merged**